i2c/common/libi2c-msg.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i2c msg client library
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marek Bialowas
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/msg.h>
+#include <sys/types.h>
+
+#include <stdio.h>
+#include <errno.h>
+
+#include <i2c.h>
+#include <i2c-msg.h>
+
+static struct {
+	oid_t i2c_oid;
+	int initialized;
+} common;
+
+/* opens the device */
+int i2c_init(unsigned int dev_no)
+{
+	char devname[sizeof("/dev/i2cX")];
+	int res = 0;
+
+	sprintf(devname, "/dev/i2c%u", dev_no % 10);
```

Comment:
Maybe instead `sprintf()`:
```suggestion
	char devname[] = "/dev/i2cX";
	int res = 0;

	devname[8] = '0' + (dev_no % 10);
```
it will be faster.

Reply:
That's what I was going to suggest ;)

Reply:
This is also less error prone - string is in one place.